src/stepfunctions/steps/compute.py

```diff
@@ -38,9 +39,9 @@ def __init__(self, state_id, wait_for_callback=False, **kwargs):
             output_path (str, optional): Path applied to the state’s output after the application of `result_path`, producing the effective output which serves as the raw input for the next state. (default: '$')
         """
         if wait_for_callback:
-            kwargs[Field.Resource.value] = 'arn:aws:states:::lambda:invoke.waitForTaskToken'
+            kwargs[Field.Resource.value] = 'arn:' + get_aws_partition() + ':states:::lambda:invoke.waitForTaskToken'
```

Comment:
_Not necessarily blocking:_ I would like to at least consider class for ARN construction instead of building strings with repeated calls to `get_aws_partition()` inline everywhere. The AWS SDK in other languages have utilities for this, but python doesn't.

Something like:
```python
kwards[Field.Resource.value] = ARN(service='states', resource='lambda:invoke')
```

Or even scoping it to service integration ARNs:
```python
kwargs[Field.Resource.value] = ServiceIntegrationARN(service='lambda', api='invoke', integration_pattern=IntegrationPattern.WAIT_FOR_TASK_TOKEN)
```

Reply:
This will definitely make the code cleaner. I will look into it.
If I can not finish this quickly I will prepare a PR specific for it later.